surveys/views.py

```diff
@@ -0,0 +1,75 @@
+from django.contrib.messages import success
+from django.template.response import TemplateResponse
+from django.shortcuts import get_object_or_404
+from django.http import Http404
+from django.urls import reverse_lazy
+from django.utils.timezone import now
+from django.views.generic import FormView, ListView
+from django.views.generic.detail import SingleObjectMixin
+
+from surveys.forms import SurveyForm
+from surveys.models import Survey, Response
+
+
+class SurveyOverview(ListView):
+    template_name = "surveys/survey_overview.html"
+    model = Survey
+    context_object_name = 'surveys'
+
+
+class SurveyFormView(SingleObjectMixin, FormView):
+    template_name = "surveys/survey_form_page.html"
+    form_class = SurveyForm
+    success_url = reverse_lazy('surveys:overview')
+    model = Survey
+    pk_url_kwarg = 'survey_id'
+    context_object_name = 'survey'
+
+    def setup(self, request, *args, **kwargs):
+        super(SurveyFormView, self).setup(request, *args, **kwargs)
+        self.object = self.get_object()
+
+    def dispatch(self, request, *args, **kwargs):
+        if self.object.end_date is not None and self.object.end_date < now():
+            return self.render_expired_response()
```

Comment:
Could pass on the request to a different View (inheriting from TemplateResponse) instead.